symengine/polys/cancel.h

```diff
@@ -0,0 +1,24 @@
+
+#ifndef SYMENGINE_CANCEL_H
+#define SYMENGINE_CANCEL_H
+
+#include <symengine/basic.h>
+#include <symengine/polys/basic_conversions.h>
+
+#ifdef HAVE_SYMENGINE_FLINT
+using SymEngine::UIntPolyFlint;
+
+using SymEngine::RCP;
+using SymEngine::divides_upoly;
+
+namespace SymEngine
+{
+// Declaration of cancel function
+void cancel(const RCP<const Basic> &numer, const RCP<const Basic> &denom,
+            const Ptr<RCP<const UIntPolyFlint>> &result_numer,
```

Comment:
Can you make this Poly type a template, so that it's not specific to flint type?

Reply:
Yaa sure!!

Reply:
```
/home/travis/build/symengine/symengine/symengine/tests/polynomial/test_cancel.cpp:27: undefined reference to `void SymEngine::cancel<SymEngine::UIntPolyFlint>(Teuchos::RCP<SymEngine::Basic const> const&, Teuchos::RCP<SymEngine::Basic const> const&, Teuchos::Ptr<Teuchos::RCP<SymEngine::UIntPolyFlint const> > const&, Teuchos::Ptr<Teuchos::RCP<SymEngine::UIntPolyFlint const> > const&, Teuchos::Ptr<Teuchos::RCP<SymEngine::UIntPolyFlint const> > const&)
```
This error is being shown.

Reply:
Since this is a template function, the implementation has to be in the header file with the `inline` keyword added to the function signature. You can delete `cancel.cpp` then.

Reply:
Done @isuruf